org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/TestExecutorResource.xtend

```diff
@@ -0,0 +1,37 @@
+package org.testeditor.web.backend.persistence
+
+import javax.inject.Inject
+import javax.ws.rs.Consumes
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.UriBuilder
+import org.slf4j.LoggerFactory
+import javax.ws.rs.Produces
+
+@Path("/tests")
+@Consumes(MediaType.APPLICATION_JSON)
+class TestExecutorResource {
+
+	static val logger = LoggerFactory.getLogger(TestExecutorResource)
+
+	@Inject TestExecutorProvider executorProvider
+
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.TEXT_PLAIN)
+	@Path("execute") 
+	def Response executeTests(@QueryParam("resource") String resourcePath) {
+		val builder = executorProvider.testExecutionBuilder(resourcePath)
+		if (builder !== null) {
```

Comment:
I think if you throw exceptions in `TestExecutorProvider.testExecutionBuilder`, you will not only be able to drop the null test, but can also rely on Jersey to take care of responding with a server error that should contain the exception message and associated stack trace.